test/oncluster/scenario_github_test.go

```diff
@@ -0,0 +1,90 @@
+//go:build oncluster
+// +build oncluster
+
+package oncluster
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"gotest.tools/v3/assert"
+	common "knative.dev/func/test/common"
+)
+
+/*
+Test scenario covered here:
+ - As a Developer I want my function stored on a public GitHub repo to get deployed on my cluster
+
+Notes:
+ * The function used as input for this scenario is stored in this repository at /test/oncluster/testdata/simplefunc
+
+ * On a CI Pull Request action (env CI="true") the branch used on the on-cluster test is the pull request reference.
+   The equivalent deploy func command would look like this:
+
+   func deploy --remote \
+     --git-url https://github.com/knative/func \
+     --git-dir test/oncluster/testdata/simplefunc \
+     --git-branch refs/pull/1650/head
+
+ * When not on CI,  the branch used will be "main" and the repository is https://github.com/knative/func.git
+
+*/
+
+func resolveGitVars() (gitRepoUrl string, gitRef string) {
+	var githubRepository = "knative/func"
+	gitRef = "main"
+	if ci, _ := os.LookupEnv("CI"); ci == "true" {
```

Comment:
If you are not interested in `"ok"` value you can use "os.Getenv()".

Reply:
sure I'll push a fix on that